examples/http-certification/upgrade-to-update-call/README.md

```diff
@@ -0,0 +1,134 @@
+# Upgrading HTTP calls to update calls
+
+## Overview
+
+This guide walks through an example project that demonstrates how to use the ["Upgrade to Update call"](https://internetcomputer.org/docs/current/references/http-gateway-protocol-spec#upgrade-to-update-calls) feature of the HTTP Gateway.
+
+Since browsers are unable to directly interact with the IC network, the HTTP Gateway acts as a bridge between the two. The HTTP Gateway forwards requests from clients to canisters and forwards responses from canisters back to clients. Before returning responses from canister back to clients, the HTTP Gateway verifies the certification of the response to ensure that they have not been tampered with.
+
+Upgrading query calls to upgrade calls allows for the certification of any kind of dynamic response by leveraging ICP's consensus protocol without having to statically certify the response ahead of time. This is the simplest way to add _secure_ HTTP support to a canister.
+
+A similairly simple yet more performant, but _insecure_ approach is to skip certification entirely. This is not recommended unless you are absolutely sure that certification really does not make sense for your canister. Check the ["Skipping certification for HTTP responses"](https://internetcomputer.org/docs/current/developer-docs/web-apps/http-compatible-canisters/skipping-certification-for-http-responses) guide for more details on how to do that.
+
+This is not a beginner's canister development guide. Many foundational concepts that a relatively experienced canister developer should already know will be omitted. Concepts specific to upgrading to an update call will be called out here and can help to understand the [full code example](https://github.com/dfinity/response-verification/tree/main/examples/http-certification/upgrade-to-update-call).
```

Comment:
nit: I'd use "fundamental" rather than "foundational" (but I'm not a native speaker :wink:)

Reply:
Interesting, I didn't think about this when I wrote it. But I asked ChatGPT for advice and it gave me the following:

The distinction between "foundational concepts" and "fundamental concepts" is subtle, but they can be understood in different contexts:

1. **Foundational Concepts**:
   - **Role**: These are concepts that serve as the base or groundwork for a system of thought, discipline, or framework. They help build more complex ideas and are often specific to a certain field or structure. 
   - **Scope**: These concepts may not be universally applicable across all domains but are essential for understanding a particular area or system.
   - **Example**: In mathematics, the concept of "number" or "set" is foundational because it's critical to build the rest of the discipline.

2. **Fundamental Concepts**:
   - **Role**: These refer to basic, core ideas that are essential and universal to understanding broader or multiple systems of thought. They are more intrinsic and represent essential truths or properties that underlie a field.
   - **Scope**: Fundamental concepts often have broad applicability and are seen as essential truths or principles across different disciplines.
   - **Example**: In physics, the concept of "energy" or "force" is fundamental because it applies universally across many different branches of physics.

### Key Difference:
- **Foundational concepts** are more about what forms the basis for further learning or development in a particular area.
- **Fundamental concepts** are about what is universally necessary or basic across multiple systems or domains.

In practice, people may use these terms interchangeably, but the emphasis can shift based on whether they are talking about a specific system (foundational) or universal ideas (fundamental).

Reply:
I updated it to fundamental anyway, I think it sounds nicer to say